components/table/SelectionCheckboxAll.tsx

```diff
@@ -151,36 +152,45 @@ export default class SelectionCheckboxAll extends React.Component<SelectionCheck
   }
 
   render() {
-    const { disabled, prefixCls } = this.props;
+    const { disabled, prefixCls, selections } = this.props;
     const { checked, indeterminate } = this.state;
 
     let selectionPrefixCls = `${prefixCls}-selection`;
 
-    let selections = this.defaultSelections.concat(this.props.selections || []);
+    let customSelections: React.ReactNode = null;
 
-    let menu = (
-      <Menu
-        className={`${selectionPrefixCls}-menu`}
-        selectedKeys={[]}
-      >
-        {this.renderMenus(selections)}
-      </Menu>
-    );
+    if (selections) {
+      let newSelections = Array.isArray(selections) ? this.defaultSelections.concat(selections)
+      : this.defaultSelections;
+
+      let menu = (
+        <Menu
+          className={`${selectionPrefixCls}-menu`}
+          selectedKeys={[]}
+        >
+          {this.renderMenus(newSelections)}
+        </Menu>
+      );
+
+      customSelections =  (
+        <Dropdown overlay={menu}>
+          <div className={`${selectionPrefixCls}-down`}>
+            <Icon type="down" />
+          </div>
+        </Dropdown>
+      );
+    }
 
     return (
       <div className={selectionPrefixCls}>
         <Checkbox
-          className={`${selectionPrefixCls}-select-all`}
+          className={classNames(`${selectionPrefixCls}-select-all`, {custom: customSelections})}
```

Comment:
1. custom => ${selectionPrefixCls}-select-custom
2. should have whitespaces inside `{}`